src/library/browse/foldertreemodel.cpp

```diff
@@ -48,26 +46,33 @@ bool FolderTreeModel::hasChildren(const QModelIndex& parent) const {
 }
 
 bool FolderTreeModel::directoryHasChildren(const QString& path) const {
-    auto it = m_directoryCache.constFind(path);
-    if (it != m_directoryCache.constEnd()) {
-        return it.value();
+    bool ignore;
+    for (const auto& ignorePath : std::as_const(m_cacheIgnoreList)) {
+        if (path.startsWith(ignorePath)) {
+            ignore = true;
+            break;
+        }
+    }
```

Comment:
Apparently the cache has been  implemented to boost performance for dirs with many children QStringList and startsWith() _may_ not be the best choice here.

I'll look into [QStringMatcher](https://doc.qt.io/qt-6/qstringmatcher.html) and check for oher performant solutions.